spring-integration-mongodb/src/main/java/org/springframework/integration/mongodb/store/ConfigurableMongoDbMessageStore.java

```diff
@@ -203,12 +204,26 @@ public MessageGroup removeMessageFromGroup(final Object groupId, final Message<?
 		Assert.notNull(groupId, "'groupId' must not be null");
 		Assert.notNull(messageToRemove, "'messageToRemove' must not be null");
 
+		doRemove(groupId, messageToRemove);
+		updateGroup(groupId, lastModifiedUpdate());
+		return getMessageGroup(groupId);
+	}
+
+	protected void doRemove(final Object groupId, final Message<?> messageToRemove) {
 		Query query = groupIdQuery(groupId)
 				.addCriteria(Criteria.where(MessageDocumentFields.MESSAGE_ID).is(messageToRemove.getHeaders().getId()));
 		mongoTemplate.remove(query, collectionName);
-		updateGroup(groupId, lastModifiedUpdate());
-		return getMessageGroup(groupId);
+	}
+
+	@Override
+	public void removeMessagesFromGroup(Object groupId, Collection<Message<?>> messages) {
+		Assert.notNull(groupId, "'groupId' must not be null");
+		Assert.notNull(messages, "'messageToRemove' must not be null");
 
+		for (Message<?> messageToRemove : messages) {
+			doRemove(groupId, messageToRemove);
```

Comment:
Here you are: https://www.compose.io/articles/better-bulking-for-mongodb-2-6-and-beyond/ !
So, we just need to study how to use `this.mongoTemplate.getCollection(this.collectionName).initializeOrderedBulkOperation()`